src/deepsparse/pipeline.py

```diff
@@ -185,10 +198,19 @@ def __call__(self, *args, monitoring: bool = False, **kwargs) -> BaseModel:
             )
         timer = TimingBuilder()
 
-        # parse inputs into input_schema
         timer.start(InferencePhases.TOTAL_INFERENCE)
+
+        # ------ PREPROCESSING ------
```

Comment:
Personally found it useful to visually separate the __call__() into three parts: preprocessing, inference and postprocessing.